### ISSUE TITLE
Standard initialization of storage back-ends

### ISSUE THREAD

**Issue by kontsevoy** [gravitational/teleport#688]
TL;DR This ticket calls for standardization of plug-in utilization using `etcd` implementation as a template.

## Problem

Currently our storage back-ends all implement a nifty small interface with a well-defined behavior. But the creation & configuration of back-ends is not standardized at all, so we have utility functions like `initBoltBk()` or `initEtcdBk()` or `initDynamo()` _outside_ of `lib/backend/xxx` package, and _inside_ of Teleport's core config parsing. These functions are a bit crazy too: there are 2 "config" structures in existence per back-end: one for YAML parsing and another for JSON.

## Solution

Introduce a new API function every back-end _must_ implement:

```go
type backend.Params map[string]interface{}
func backend.New(params backend.Params) (backend.Backend, error)
```

So every back-end will have to implement two things:
* The `backend.Backend` interface
* The `backend.New()` function

## Implementation

Teleport core will read `storage` section in YAML and will construct a property bag of `backend.Params` type. This property bag will be passed to `backend.New()` which every back-end implements. 

A back-end will have its own (non-exported) Config structure (they already do) with proper `json:xxx` serialization attributes and they can simply call `utils.ObjectToStruct()` to convert `backend.Params` to their own internal Config.

From a user perspective, back-end will be controlling every value under `storage:` YAML section, just like etcd and dynamo already do.

## Side Benefits

* This will kill a lot of spaghetti code which exists right now across dynamo/etcd/bolt/fs
* It will be trivial to implement back-ends as dynamically loaded `.so` plug-ins.
